spiffworkflow-backend/src/spiffworkflow_backend/services/message_service.py

```
@@ -158,7 +158,12 @@ def correlate_send_message(
                 message_instance_receive.failure_cause = str(exception)
                 db.session.add(message_instance_receive)
             if processor_receive is not None:
-                processor_receive.save()
+                # We may not be able to save here, this can raise a new exception.
+                try:
+                    processor_receive.save()
+                except Exception:
+                    db.session.commit()
+                    raise exception
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)